--- a/net/minecraft/world/storage/MapData.java
+++ b/net/minecraft/world/storage/MapData.java
@@ -209,15 +209,15 @@
       int i = 1 << this.field_76197_d;
       float f = (float)(p_191095_4_ - (double)this.field_76201_a) / (float)i;
       float f1 = (float)(p_191095_6_ - (double)this.field_76199_b) / (float)i;
-      byte b0 = (byte)((int)((double)(f * 2.0F) + 0.5D));
-      byte b1 = (byte)((int)((double)(f1 * 2.0F) + 0.5D));
+      byte b0 = (byte)MathHelper.func_76125_a((int)((double)(f * 2.0F) + 0.5D), Byte.MIN_VALUE, Byte.MAX_VALUE);
+      byte b1 = (byte)MathHelper.func_76125_a((int)((double)(f1 * 2.0F) + 0.5D), Byte.MIN_VALUE, Byte.MAX_VALUE);
       int j = 63;
       byte b2;
-      if (f >= -63.0F && f1 >= -63.0F && f <= 63.0F && f1 <= 63.0F) {
+      if (f > -64.0F && f1 > -64.0F && f < 64.0F && f1 < 64.0F) {
          p_191095_8_ = p_191095_8_ + (p_191095_8_ < 0.0D ? -8.0D : 8.0D);
          b2 = (byte)((int)(p_191095_8_ * 16.0D / 360.0D));
-         if (this.field_76200_c == DimensionType.field_223228_b_ && p_191095_2_ != null) {
-            int l = (int)(p_191095_2_.func_72912_H().func_76073_f() / 10L);
+         if (p_191095_2_ != null && p_191095_2_.func_201672_e().field_73011_w.shouldMapSpin(p_191095_3_, p_191095_4_, p_191095_6_, p_191095_8_)) {
+            int l = (int)(p_191095_2_.func_201672_e().func_72820_D() / 10L);
             b2 = (byte)(l * l * 34187121 + l * 121 >> 15 & 15);
          }
       } else {
@@ -239,19 +239,19 @@
          }
 
          b2 = 0;
-         if (f <= -63.0F) {
+         if (f < -64.0F) {
             b0 = -128;
          }
 
-         if (f1 <= -63.0F) {
+         if (f1 < -64.0F) {
             b1 = -128;
          }
 
-         if (f >= 63.0F) {
+         if (f > 64.0F) {
             b0 = 127;
          }
 
-         if (f1 >= 63.0F) {
+         if (f1 > 64.0F) {
             b1 = 127;
          }
       }
